feature-libs/organization/administration/components/shared/list/list.component.html

```diff
@@ -54,10 +54,9 @@ <h3>
           <a
             *ngIf="!hideAddButton"
             class="button primary create"
-            [routerLink]="{ cxRoute: structure.type + 'Create' } | cxUrl"
-            routerLinkActive="disabled"
+            (click)="onAddButtonClick()"
           >
-            {{ 'organization.add' | cxTranslate }}
+            {{ getAddButtonLabel() | cxTranslate }}
```

Comment:
Regarding this [recent commit](https://github.com/SAP/spartacus/pull/16748/commits/76d47b20405bebce0aeb4a9ea6e30595d25ceba3) , I’m afraid it doesn’t help to avoid copying big portions of code It just moved some stuff from template to TS of the component. But only moving logic to an extendable component service could help more in extending this logic with your CDC lib, but keeping the old original component.

btw. this also introduced 2 flaws:
- potential memory leak -> we create an unhandled subscription in TS class
- a11y degradation -> we have a <a> element without href, but only onClick handler